vivisect/analysis/generic/symswitchcase.py

```diff
@@ -0,0 +1,1247 @@
+
+            if sz == self.vw.psize and self.vw.isValidPointer(tgt):
+                for count in range(upper-lower+1):
+                    self.vw.makePointer(addr + (sz*count))
+
+            else:
+                for count in range(upper-lower+1):
+                    self.vw.makeNumber(addr + (sz*count), sz)
+
+    def iterCases(self):
+        '''
+        Generator which yields the case index and the target address
+        '''
+        vw = self.vw
+
+        (csemu, cseffs), aspath, fullpath = self.getSymbolikParts()
+
+        symidx = self.getSymIdx()
+        logger.info('getSymIdx: %r', symidx)
+
+        jmptgt = self.getSymTarget()
+        lower, upper, offset = self.getBounds()
+        logger.debug( " itercases: %r %r %r %r", symidx, lower, upper, offset)
+
+        # use a TrackingSymbolikEmulator so we get the "readSymMemory()" function actually hitting 
+        # the r/o data from our VivWorkspace
+        symemu = TrackingSymbolikEmulator(vw)
+        symemu.setSymSnapshot(csemu.getSymSnapshot())
+
+        symemu.setSymVariable(symidx, None) # don't want to update this out of the symbolik state
+        #workJmpTgt = jmptgt.update(emu=symemu)
+
+
+        for idx in range(lower-offset, upper-offset+1):
+            symemu.setSymVariable(symidx, Const(idx, 8))
+            workJmpTgt = jmptgt.update(emu=symemu)
+            logger.info(" itercases: workJmpTgt: %r", workJmpTgt)
+            coderef = workJmpTgt.solve(emu=symemu, vals={symidx:idx})
+            #coderef = jmptgt.update(emu=symemu)
+            logger.debug(" itercases: %r, %r", idx, hex(coderef))
+            yield idx + offset, coderef
+
+def makeNames(vw, jmpva, cases, baseoff=0):
+    '''
+    Create names and xrefs for each identified case.
+    '''
+    for addr, indexes in cases.items():
+        outstrings = []
+        combined = False
+        start = last = UNINIT_CASE_INDEX
+        for case in indexes:
+            if case == last+1:
+                if not combined:
+                    combined = True
+                    start = last
+            else:
+                if combined:
+                    combined = False
+                    outstrings.append("%Xto%X" % (start+baseoff, last+baseoff))
+                elif last != UNINIT_CASE_INDEX:
+                    outstrings.append("%X" % (last+baseoff))
+            last = case
+
+        # catch the last one if highest cases are combined
+        if combined:
+            combined = False
+            outstrings.append("%Xto%X" % (start+baseoff, last+baseoff))
+        else:
+            outstrings.append("%X" % (case+baseoff))
+
+        logger.info("OUTSTRINGS: %s", repr(outstrings))
+
+        idxs = '_'.join(outstrings)
+        casename = "case_%s_%x" % (idxs, addr)
+
+        curname = vw.getName(addr) 
+        if curname is not None:
+            ## FIXME NOW!:   if already labeled, chances are good this is other cases in the same function.
```

Comment:
Sounds urgent

Reply:
yeah, it does.  but no fix is obvious.  and it was just piss and wind.